presto-iceberg/src/main/java/com/facebook/presto/iceberg/TableStatisticsMaker.java

```diff
@@ -229,9 +254,11 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle, Cons
         TableStatistics.Builder result = TableStatistics.builder();
         result.setRowCount(Estimate.of(recordCount));
         result.setTotalSize(Estimate.of(summary.getSize()));
+        Map<Integer, ColumnStatistics.Builder> tableStats = getMostRecentStatisticsFileForTableSnapshot(tableHandle)
+                .map(TableStatisticsMaker::loadStatisticsFile).orElseGet(Collections::emptyMap);
```

Comment:
Is there a better way to determine which statistics snapshot to use? In a specified snapshot read, when there do not exists statistics on the target snapshot, sometimes maybe it's better to use the statistics after the target snapshot. Just for discuss.

Reply:
I've updated the logic to prefer using the snapshot with the smallest absolute difference in time between the target snapshot and statistics file, regardless if the file was generated before or after the snapshot